src/test/java/org/jabref/rest/jabmap/resources/RootResourceTest.java

```diff
@@ -0,0 +1,479 @@
+package org.jabref.rest.jabmap.resources;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.InternalField;
+import org.jabref.model.entry.field.MindMapField;
+import org.jabref.model.entry.types.MindMapEntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+import org.jabref.model.jabmap.EdgeDirection;
+import org.jabref.model.jabmap.MindMapEdge;
+import org.jabref.model.jabmap.MindMapNode;
+import org.jabref.rest.jabmap.JabMapHTTPServer;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class RootResourceTest {
+
+    private static final String WEB_SERVICE_URI = "http://localhost:9898/";
+    private static Client client;
+
+    @BeforeAll
+    static void setUp() {
+        JabMapHTTPServer.startHttpEndPoint();
+        client = ClientBuilder.newClient();
+    }
+
+    @AfterAll
+    static void close() {
+        client.close();
+    }
+
+    @BeforeEach
+    void initMockDatabase() {
+        RootResource.databaseAccess = new MockDatabaseAccess();
+    }
+
+    @Test
+    void getReferenceEntries() {
+        List<BibEntry> bibEntries = setUpEntriesForGetReferenceEntries();
+        RootResource.databaseAccess.getActiveDatabase().insertEntries(bibEntries);
+
+        Response response = null;
+        try {
+            response = client.target(WEB_SERVICE_URI).path("libraries/current/entries").request().accept(MediaType.APPLICATION_JSON).get();
+            assertEquals(200, response.getStatus());
+            String responseBody = response.readEntity(String.class);
+            String expectedJson = "[{\"type\":{\"bibtex_metadata\":\"Article\",\"key\":\"article1\"}},{\"type\":{\"bibtex_metadata\":\"Book\",\"key\":\"book1\"}},{\"type\":{\"bibtex_metadata\":\"MastersThesis\",\"key\":\"MastersThesis1\"}}]";
```

Comment:
This weill be a maintenance hell, because the whitespace and key ordering does not matter for equality.

I think, JSONassert should be used. --> https://github.com/skyscreamer/JSONassert. Then only one line needs to be changed --> `assertEquals` --> `JSONAssert.assertEquals`.

Maybe try with Jackson or another JSON library for a equality? --> https://www.baeldung.com/jackson-compare-two-json-objects

It does not seem possible with REST-assured to check for the complete json directly.